types/blob/share_builder.go

```diff
@@ -0,0 +1,228 @@
+package blob
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/rollkit/celestia-openrpc/types/appconsts"
+	"github.com/rollkit/celestia-openrpc/types/namespace"
+	"github.com/rollkit/celestia-openrpc/types/share"
+)
+
+type Builder struct {
+	namespace      namespace.Namespace
+	shareVersion   uint8
+	isFirstShare   bool
+	isCompactShare bool
+	rawShareData   []byte
+}
+
+func NewEmptyBuilder() *Builder {
+	return &Builder{
+		rawShareData: make([]byte, 0, appconsts.ShareSize),
+	}
+}
+
+// Init() needs to be called right after this method
+func NewBuilder(ns namespace.Namespace, shareVersion uint8, isFirstShare bool) *Builder {
+	return &Builder{
+		namespace:      ns,
+		shareVersion:   shareVersion,
+		isFirstShare:   isFirstShare,
+		isCompactShare: isCompactShare(ns),
+	}
+}
+
+func (b *Builder) Init() (*Builder, error) {
+	if b.isCompactShare {
+		if err := b.prepareCompactShare(); err != nil {
+			return nil, err
+		}
+	} else {
+		if err := b.prepareSparseShare(); err != nil {
+			return nil, err
+		}
+	}
+
+	return b, nil
+}
```

Comment:
Why are we returning the `Builder`? It isn't a new builder and the methods are just modifying the original?

Can't we just do this?

```suggestion
func (b *Builder) Init() (error) {
	if b.isCompactShare {
		return b.prepareCompactShare()
	} 
	return b.prepareSparseShare()
}
```

Reply:
Follow up: https://github.com/celestiaorg/celestia-app/issues/2157